tests/yast2_gui/yast2_expert_partitioner.pm

```diff
@@ -0,0 +1,146 @@
+# - Inside "vgtest", create lv4, type raw
+# - Start xterm, run "lvdisplay /dev/vgtest/lv<number>" for each partition
+# - Close xterm, start a new yast2 storage and delete all partitions created
+#
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+use base "y2_module_guitest";
+use strict;
+use warnings;
+use testapi;
+use YuiRestClient;
+use scheduler 'get_test_suite_data';
+
+my $partitioner;
+my $test_data;
+
+sub pre_run_hook {
+    $partitioner = $testapi::distri->get_expert_partitioner();
+    $test_data   = get_test_suite_data();
+}
+
+sub open_expert_partitioner {
+    my ($self) = shift;
+    $self->launch_yast2_module_x11('storage', extra_vars => get_var('YUI_PARAMS'));
+    YuiRestClient::connect_to_app_running_system();
+    $partitioner->get_confirmation_warning_controller()->confirm_only_use_if_familiar();
+}
+
+sub add_custom_partition {
+    my $disk = $test_data->{disks}[0];
+    $partitioner->add_partition_on_gpt_disk({
+            disk      => $disk->{name},
+            partition => $disk->{partitions}[0]
+    });
+    $partitioner->show_summary_and_accept_changes();
+}
+
+sub verify_custom_partition {
+    my $index     = shift // 0;
+    my $partition = $test_data->{disks}[0]->{partitions}[$index];
+    my $name      = "/dev/$partition->{name}";
+    my $size      = $partition->{size} =~ s/iB//r;
+
+    x11_start_program('xterm');
+    become_root;
+    wait_still_screen 3;
+    validate_script_output("fdisk -l | grep $name",
+        sub { m/$name\s+\d+\s+\d+\s+\d+\s+$size.*/ });
+    send_key "ctrl-d";
+    wait_screen_change { send_key "ctrl-d" };
+}
+
+sub resize_custom_partition {
+    $partitioner->resize_partition({
+            disk      => $test_data->{disks}[0]->{name},
+            partition => $test_data->{disks}[0]->{partitions}[1]
+    });
+    $partitioner->show_summary_and_accept_changes();
+}
+
+sub verify_resized_partition {
+    verify_custom_partition(1);
+}
+
+sub delete_resized_partition {
+    $partitioner->delete_partition({
+            disk      => $test_data->{disks}[0]->{name},
+            partition => $test_data->{disks}[0]->{partitions}[1]
+    });
+    $partitioner->show_summary_and_accept_changes();
+}
+
+sub add_logical_volumes {
+    $partitioner->setup_lvm($test_data->{lvm});
+    $partitioner->show_summary_and_accept_changes();
+}
+
+sub verify_logical_volumes {
+    x11_start_program('xterm');
+    become_root;
+    wait_still_screen 3;
+    my $vg = $test_data->{lvm}->{volume_groups}[0];
+    assert_script_run "lvdisplay /dev/$vg->{name}/$_->{name}" for @{$vg->{logical_volumes}};
+    send_key "ctrl-d";
+    wait_screen_change { send_key "ctrl-d" };
+}
+
+sub delete_volume_group {
+    my $vg = $test_data->{lvm}->{volume_groups}[0];
+    $partitioner->delete_volume_group($vg->{name});
+    $partitioner->show_summary_and_accept_changes();
+}
+
+sub run {
+    my ($self) = shift;
+    select_console "x11";
+
+    $self->open_expert_partitioner;
```

Comment:
I was thinking about moving this part all together to `add_custom_partition`, as initially I was confused where we close expert partitioner. It would be more consistent if we would either open it here and also call `show_summary_and_accept_changes`, so it's clear that we have opened it, added partition and closed. Or move it all in `add_custom_partition`, so we can see that test adds partition and then validate it.
Same applied to other checks below.